### DISCUSSION TITLE
Backup with borgbackup excluding the mounted / binded datadir

### DISCUSSION THREAD

**Discussion by monsieurhannes** [nextcloud/all-in-one#2168] [Ideas] [answered]
Hi,

Is it possible to backup the containers, database and so on, without including the datadir that in my case is binded and has its own legacz borg-backup?
I'm moving a part of my docker services to another computer and nextcloud is among those...

Thanks in advance!

**Comment by szaimen** [comment, marked as answer]
Hi, I think you are most likely looking for this: https://github.com/nextcloud/all-in-one#how-to-enable-automatic-updates-without-creating-a-backup-beforehand as you are using your own backup solution. There is indeed no way to exclude the datadir from the backup and we also don't want to add an option for this.

**Comment by BastienX8** [comment]
Hello,
I recently moved to Nextcloud AIO and everything works fine so far, but the current backup solution is not ideal. I have over 1TB of data stored on 2x 2TB in raid, mounted using the NEXTCLOUD_DATADIR environment variable. With the current solution the backups would be **huge**, take a long time to complete and wear the drives quickly...
Like a lot of people I use the environment variable to handle the big datadir separately, but everything else is still managed by the containers. It would be really easier to maintain if there was a way to automatically backup and restore the system without needing an additional 10TB drive. Isn't there a way to do something like that ?

**Reply by szaimen** [reply]
See https://github.com/nextcloud/all-in-one#how-to-enable-automatic-updates-without-creating-a-backup-beforehand

**Reply by BastienX8** [reply]
Thanks for the quick reply, but this doesn't solve the backup problem because we still have to search in every containers where the data are stored (and hope to not break everything if we need to restore the system), that's why you created a backup solution with borg. Isn't there a way to exclude the datadir from the backup system, or at least the user files ? Because currently it is not usable for all those who have a lot of data

**Reply by szaimen** [reply]
All important aio data except nextclouds datadir if configured will be stored in docker volumes so you don't need to search for every container manually.

**Reply by BastienX8** [reply]
Ok, and if we need to restore or reinstall the system at some point do I have to proceed in a specific order to put the datas back in place ? For example if I reinstall nextcloud aio is it better to create the corresponding volumes beforehand, or let it create all volumes by itself and replace them afterwards ?

**Reply by szaimen** [reply]
> is it better to create the corresponding volumes beforehand, or let it create all volumes by itself and replace them afterwards ?

both should be fine

**Reply by BastienX8** [reply]
Thanks !

**Reply by BastienX8** [reply]
Just by curiosity, why not adding an option to exclude the datadir in the backup system ?

**Reply by szaimen** [reply]
> Just by curiosity, why not adding an option to exclude the datadir in the backup system ?

Because you will always need to restore both the database and the datadir (and all other data like the FTS metadata) at the same time which wen cannot make sure if we are not including the datadirectory. Otherwise the database and your data (and other metadata like FTS) will get out of sync. A scan could be scheduled but will not work in all cases, e.g. when encryption is enabled and would take a long time if the data is huge. Also in the datadir included is the appdata directory which also need to get restored at the same time like all the other data.

**Reply by BastienX8** [reply]
The system could check if the backup contains a datadir or not, and if the env. variable for the datadir is used or not, and only allows you to restore the selected backup if the env. variable is set.

**Reply by szaimen** [reply]
I think you did not understand the problem, did you? the problem is that we need to restore everything at the same time including the datadir because otherwise things get problematic. So you would need tomake sure yourself that this is the case when the datadir is not included which we cannot check for. So better to advise people to just use their own backup solution for the whole thing in that case. I've seen to many cases where people don't read the documentation. So adding an option for people to break their instance upon restore is not going to be implemented from my side.

**Reply by BastienX8** [reply]
Ok it is clearer now. It's just that it would have been a nice bonus for people who have a lot of data to be able to manage the backups from the aio admin page, which is of course not possible with our own backup solution. But I guess at this point I could also make my own admin page.
